src/core/lib/iomgr/endpoint_pair_windows.cc

```diff
@@ -63,8 +63,8 @@ static void create_sockets(SOCKET sv[2]) {
   GPR_ASSERT(svr_sock != INVALID_SOCKET);
 
   closesocket(lst_sock);
-  grpc_tcp_prepare_socket(cli_sock);
-  grpc_tcp_prepare_socket(svr_sock);
+  GPR_ASSERT(grpc_tcp_prepare_socket(cli_sock).ok());
+  GPR_ASSERT(grpc_tcp_prepare_socket(svr_sock).ok());
```

Comment:
From the test failure it looks like the original code does not expect these calls to succeed. `grpc_tcp_prepare_socket` does 3 things:
  1. set socket to nonblocking
  2. set an IPv6 socket to dualstack
  3. enable TCP_NODELAY on the socket

I think for 2, we need to create IPv6 sockets first and this option should be set before `bind()` for both the server's listening socket and the client socket. This function creates IPv4 sockets so this setting will fail and it's also setting it on a connected server socket (which may also fail).

Given this, I suggest we log the error here in case we want to deal with it later with something like:
```
  grpc_error_handle error = grpc_tcp_prepare_socket(cli_sock);
  if (!error.ok()) {
    gpr_log(GPR_ERROR, "Prepare cli_sock failed with error: %s",
            grpc_core::StatusToString(error).c_str());
  }
  error = grpc_tcp_prepare_socket(svr_sock);
  if (!error.ok()) {
    gpr_log(GPR_ERROR, "Prepare svr_sock failed with error: %s",
            grpc_core::StatusToString(error).c_str());
  }
```

CC @drfloob who may have some thoughts on this.

Reply:
Oops, AJ is faster..